hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -267,9 +277,105 @@ spec = describe "ServerSpec" $
+      showLogsOnFailure $ \tracer ->
+        withFreePort $ \port -> do
+          withAPIServer @SimpleTx "127.0.0.1" port alice mockPersistence tracer noop $ \Server{sendOutput} -> do
+            withClient port "/?history=no" $ \conn -> do
+              status <- waitMatch 5 conn $ \v -> v ^? key "headStatus"
+              status `shouldBe` Aeson.String "Idle"
+
+            -- TODO: do test commit outputs as well? (or just unit/property test the projection for details)
+
+            headIsOpen <- HeadIsOpen @SimpleTx <$> generate arbitrary <*> generate arbitrary
+
+            sendOutput headIsOpen
+            withClient port "/?history=no" $ \conn -> do
+              status <- waitMatch 5 conn $ \v -> v ^? key "headStatus"
+              status `shouldBe` Aeson.String "Open"
+
+            sendOutput . ReadyToFanout $ headId headIsOpen
+            withClient port "/?history=no" $ \conn -> do
+              status <- waitMatch 5 conn $ \v -> v ^? key "headStatus"
+              status `shouldBe` Aeson.String "ClosedAfterDeadline"
+
+    it "greets with correct head status after restart" $
+      showLogsOnFailure $ \tracer ->
+        withTempDir "api-server-head-status" $ \persistenceDir ->
+          withFreePort $ \port -> do
+            apiPersistence <- createPersistenceIncremental $ persistenceDir <> "/server-output"
+            withAPIServer @SimpleTx "127.0.0.1" port alice apiPersistence tracer noop $ \Server{sendOutput} -> do
+              headIsInitializing <- generate $ HeadIsInitializing <$> arbitrary <*> arbitrary
+              sendOutput headIsInitializing
+              withClient port "/?history=no" $ \conn -> do
+                status <- waitMatch 5 conn $ \v -> v ^? key "headStatus"
+                status `shouldBe` Aeson.String "Initializing"
+
+            -- expect the api server to load events from apiPersistence and project headStatus correctly
+            withAPIServer @SimpleTx "127.0.0.1" port alice apiPersistence tracer noop $ \_ -> do
+              withClient port "/?history=no" $ \conn -> do
+                status <- waitMatch 5 conn $ \v -> v ^? key "headStatus"
+                status `shouldBe` Aeson.String "Initializing"
+
+    it "displays correctly snapshotUtxo in a Greeting message" $
+      showLogsOnFailure $ \tracer ->
+        withFreePort $ \port -> do
+          withAPIServer @SimpleTx "127.0.0.1" port alice mockPersistence tracer noop $ \Server{sendOutput} -> do
+            generatedSnapshot :: Snapshot SimpleTx <- generate arbitrary
+
+            let snapShotConfirmedMessage =
+                  SnapshotConfirmed
+                    { headId = HeadId "some-head-id"
+                    , snapshot = generatedSnapshot
+                    , signatures = mempty
+                    }
+            let expectedUtxos =
+                  toJSON (Hydra.Snapshot.utxo $ Hydra.API.ServerOutput.snapshot snapShotConfirmedMessage)
```

Comment:
Could: use pattern matches to avoid the qualified imports and leverage DuplicateRecordFields (we also prefer record field puns over using accessor functions)

```suggestion
            let SnapshotConfirmed {snapshot = Snapshot{utxo}} = snapshotConfirmedMessage 
                expectedUtxos = toJSON utxo
```

Alternatively, generate the utxo separately and just replace it in your `generatedSnapshot`.